### PR TITLE
TST: Add interface tests for pure Python DirectionGetter implementations

### PR DESCRIPTION
**Pull request by mrinalchaturvedi27**
Adds comprehensive interface tests for pure Python DirectionGetter implementations. This validates that the DirectionGetter API supports extensibility for novel tractography methods that require full Python compatibility.
Motivation
Currently, DIPY's test suite focuses on built-in direction getters (DeterministicMaximumDirectionGetter, ProbabilisticDirectionGetter, etc.). However, there's no explicit test validating that custom implementations of the DirectionGetter interface work correctly with LocalTracking.
This is important because:

Advanced users may want to implement custom tracking strategies
ML-based tractography methods (e.g., reinforcement learning, learned direction getters) require pure Python implementations for PyTorch/TensorFlow compatibility
Future work on RL-based tractography requires a validated integration approach
Documenting the expected behavior helps external contributors
Changes

New file: dipy/tracking/tests/test_direction_getter.py
Test class: PythonDirectionGetter - minimal reference implementation demonstrating pure Python DirectionGetter pattern
Test coverage (7 tests):

Basic functionality with LocalTracking
Direction normalization
In-place update semantics
Multiple tracking directions
Edge cases (zero-length vectors)
Non-identity affine transforms
Interface documentation

$ python -m pytest dipy/tracking/tests/test_direction_getter.py -v --import-mode=importlib
collected 7 items
dipy/tracking/tests/test_direction_getter.py::test_pure_python_direction_getter_basic PASSED
dipy/tracking/tests/test_direction_getter.py::test_initial_direction_shape PASSED
dipy/tracking/tests/test_direction_getter.py::test_direction_normalization PASSED
dipy/tracking/tests/test_direction_getter.py::test_inplace_direction_update PASSED
dipy/tracking/tests/test_direction_getter.py::test_multiple_tracking_directions PASSED
dipy/tracking/tests/test_direction_getter.py::test_invalid_direction_raises PASSED
dipy/tracking/tests/test_direction_getter.py::test_with_nonidentity_affine PASSED
======================================================================= 7 passed in 0.51s

$ flake8 dipy/tracking/tests/test_direction_getter.py --max-line-length=88
# No errors

### CODE REVIEW

**Comment by mrinalchaturvedi27** [comment]
Hi @skoudoro,
Thanks for the clarification, that helps a lot. I missed that DirectionGetter is deprecated, and I agree it makes sense not to add new tests around it.
I’m preparing a GSoC 2026 proposal around ML / RL-based tractography (e.g. Track-to-Learn), so I’m trying to understand DIPY’s current and future tracking architecture. To align properly, I had two quick questions:
What is the recommended replacement or direction going forward instead of DirectionGetter?
For prototyping or validating custom / learned tracking strategies (e.g. PyTorch-based), what is the preferred place or abstraction to target today- tests, examples, or documentation?
Happy to adjust direction based on project priorities. Thanks again for the feedback.

**Comment by skoudoro** [comment]
Hi @mrinalchaturvedi27 , 

you can look at deterministic or probabilistic tracking here https://github.com/dipy/dipy/blob/aaecdba94a98d0f2ad329932e7c24d31f646f0fd/dipy/tracking/tracker.py

It should give you some clue on how it works

the replacement of DirectionGetter are there: https://github.com/dipy/dipy/blob/master/dipy/tracking/propspeed.pyx 

It is missing few propagator like `eudx`, feel free to try to implement one, it should be a good exercise for understanding the framework
